### PR TITLE
Update Sony - PlayStation 2 (Redump).json

### PR DESCRIPTION
**Pull request by rufotheone-fr**
Had to separate every version of "GuitarFreaks V & DrumMania V" because Retool sorted "GuitarFreaks & DrumMania V3" as a superior version of "GuitarFreaks V2 & DrumMania V2"

### CODE REVIEW

**Review by rufotheone-fr — COMMENTED** [review]
I think that overrides don't work as intended...
Had to separate each versions of Micro Machines & NBA Street to see if this is ok.
Now waiting to test if the override of Choro Q HG is working.

**Comment by unexpectedpanda** [comment]
> I think that overrides don't work as intended...

They work as intended. There was an error in the PS2 clone list, which I've now fixed. The moral of the story is -- don't put version strings like V2 in group names and they work fine. It's in the [documentation](https://unexpectedpanda.github.io/retool/contribute-clone-lists/#overrides) (although I should fix that Mobile Suit Gundam example).

**Comment by rufotheone-fr** [comment]
It's ok for MicroMachines & NBA Street \o/
I've resolved the conflict, now I'm waiting for a merge so I can test those.

Thanks for your help and patience u_u

**Comment by unexpectedpanda** [comment]
Do you mean waiting for me to merge into main? Testing ideally happens on your local machine before that -- although now I'm curious, are you making changes directly in GitHub instead of an IDE?

If so, we can work on setting up a flow for you that'd make your life easier (although would involve learning some Git).

**Comment by rufotheone-fr** [comment]
Spotted ^_^;
Yeah I'm editing clonelists directly through github website...
I'm kinda old school and feeling more comfortable with notepad than a dedicated app (like Github Desktop).

**Comment by rufotheone-fr** [comment]
Notepad++ here _o/

**Comment by unexpectedpanda** [comment]
I'll second Andre and confirm that something like Visual Studio Code makes this much easier, and is reasonably smooth to get the hang of (and worth the time to put into, I think!).

I remember learning this stuff for the first time, and that it can be a bit overwhelming. If you're comfortable with Notepad++ and want to stick with it, do that! If the time comes when you're ready to explore other ways, I'm here to help.

Are you okay editing the clone lists directly on your own disk and testing that way? It'll help you to make more accurate submissions if you're able to test them first.

To test, there are a few things you'll want to do (I'm assuming you're more comfortable with the GUI version than command line, although the latter is faster to test with).

1.  Enable these settings in Retool (you can turn them off when you're done testing):

     debug-options

    These warnings are usually hidden from the user, as when No-Intro or Redump updates they can create a lot of noise (and there's no way I can update clone lists as fast as those groups can release DATs). The warnings tell you when something in the clone list doesn't match what's in the input DAT. Usually this is because title names have changed between DAT versions, or titles have been removed. When you're testing changes, however, they're also useful to catch where you've got names incorrect in the clone list.
    
    I usually also select **Output DAT file in legacy parent/clone format**, as this helps me to track the clone assignments more easily. Things can get a little interesting in those assignments sometimes, as the 1:1 parent/clone relationship doesn't really exist in Retool (things get especially weird with compilations). You might see some assignments that look odd at first glance, but end up perfectly fine when you realize that usually all the clones are stripped from an output DAT.

1.  Get the Compare plugin in Notepad++:

     **Plugins > Plugins Admin** > Install the Compare plugin

1.  Run Retool on the DAT that you want to edit the clone list for, to get the original output DAT. You'll compare against this later as a baseline.

1.  Make your clone list changes on your own disk (maybe back up the original first if you're not using version control), then run Retool again on the DAT.

1.  In Notepad++, open both the original output DAT and the one that was created with your modified clone list. Click one of their tabs, and click **Set as first to compare** in the menu bar.

     first-to-compare

1.  Click the second tab, and then click **Compare** to see a side-by-side view of the DATs with the differences highlighted (usually I check the `<date>` element to figure out which file is newest, although sometimes the flags in the description give it away depending on what you're testing).
     
     compare

1.  Use the down and up buttons in the same section of the menu bar to skip to the next and previous line differences.

     compare-files

1.  When you're happy with the output, then make the edit in GitHub, and copy/paste your clone list data in.

File: clonelists/Sony - PlayStation 2 (Redump).json [hunk 1]
```diff
@@ -4174,20 +4198,6 @@
...
-				{"searchTerm": "Metal Slug Complete"}
-			]
-		},
-		{
-			"group": "Metal Slug 2",
-			"compilations": [
-				{"searchTerm": "Metal Slug Anthology"},
-				{"searchTerm": "Metal Slug Complete"}
-			]
-		},
```
Comment (unexpectedpanda) [inline clonelists/Sony - PlayStation 2 (Redump).json:4190]: Don't remove the groups if the individual titles don't exist -- this is how compilations are done now. It helps the compilation selection algorithm to know what titles are in the compilation. Same for `Metal Slug X` further down.

File: clonelists/Sony - PlayStation 2 (Redump).json [hunk 2]
```diff
@@ -2503,13 +2510,12 @@
 				{"searchTerm": "GI Jockey 4 (Twin Pack)"},
 				{"searchTerm": "GI Jockey 4", "priority": 2}
 			]
-		},
+		},
```
Comment (unexpectedpanda) [inline clonelists/Sony - PlayStation 2 (Redump).json:2513]: Trailing spaces here that should be removed.

File: clonelists/Sony - PlayStation 2 (Redump).json [hunk 3]
```diff
@@ -2846,6 +2852,24 @@
 				{"searchTerm": "Guitar Hero Encore - Rocks the 80s"}
 			]
 		},
+		{
+			"group": "GuitarFreaks & DrumMania V3",
```
Comment (unexpectedpanda) [inline clonelists/Sony - PlayStation 2 (Redump).json:2856]: This won't work. The group name for both _GuitarFreaks & DrumMania V3_ and _GuitarFreaks V2 & DrumMania V2_ is `guitarfreaks & drummania`. You can see this by putting your proposed group name in the title tool. **File > Title tool**.

These are better off living in the overrides, using similar patterns to Micro Machines v4 and NBA Street V3.

File: clonelists/Sony - PlayStation 2 (Redump).json [hunk 4]
```diff
@@ -4257,6 +4260,18 @@
 				{"searchTerm": "Michigan - Report from Hell"}
 			]
 		},
+		{
```
Comment (unexpectedpanda) [inline clonelists/Sony - PlayStation 2 (Redump).json:4263]: You can remove both of these, the override for V4 alone does the job.

File: clonelists/Sony - PlayStation 2 (Redump).json [hunk 5]
```diff
@@ -4524,13 +4540,25 @@
 				{"searchTerm": "NBA 08 featuring the Life Vol. 3"}
 			]
 		},
+		{
```
Comment (unexpectedpanda) [inline clonelists/Sony - PlayStation 2 (Redump).json:4543]: You can remove this, the override for NBA Street V3 does the job.

File: clonelists/Sony - PlayStation 2 (Redump).json [hunk 6]
```diff
@@ -4524,13 +4540,25 @@
...
+			]
+		},
 		{
 			"group": "NBA Street Vol. 2",
 			"titles": [
 				{"searchTerm": "NBA Street 2 - Dunk Tengoku"},
 				{"searchTerm": "NBA Street Vol. 2"}
 			]
 		},
+		{
```
Comment (unexpectedpanda) [inline clonelists/Sony - PlayStation 2 (Redump).json:4556]: This is already in the overrides, so can be removed.

**Pull request merged**